include/pinocchio/multibody/model-graph.hpp

```diff
@@ -0,0 +1,370 @@
+//
+// Copyright (c) 2024 INRIA
+//
+
+#ifndef __pinocchio_multibody_model_graph_hpp__
+#define __pinocchio_multibody_model_graph_hpp__
+
+#include "pinocchio/multibody/fwd.hpp"
+#include "pinocchio/multibody/joint/joint-generic.hpp"
+#include "pinocchio/multibody/model.hpp"
+#include "pinocchio/spatial/inertia.hpp"
+#include "pinocchio/spatial/se3.hpp"
+
+#include <Eigen/Core>
+
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/depth_first_search.hpp>
+#include <boost/graph/visitors.hpp>
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <unordered_map>
+
+namespace pinocchio
+{
+  namespace internal
+  {
+    template<typename Derived>
+    struct JointBaseGraph
```

Comment:
I'm not sure we need CRTP here.

Also, I don't think we need inheritance here: the name is contains in ModelGraphEdge